hack/generator/pkg/astmodel/armconversion/convert_to_arm_function_builder.go

```diff
@@ -0,0 +1,558 @@
+			if !ok {
+				panic(fmt.Sprintf("Type property was not an enum, was %T", toProp.PropertyName()))
+			}
+
+			def, err := builder.codeGenerationContext.GetImportedDefinition(enumTypeName)
+			if err != nil {
+				panic(err)
+			}
+
+			enumType, ok := def.Type().(*astmodel.EnumType)
+			if !ok {
+				panic(fmt.Sprintf("Enum %v definition was not of type EnumDefinition", enumTypeName))
+			}
+
+			optionId := astmodel.GetEnumValueId(def.Name(), enumType.Options()[0])
+
+			result := astbuilder.SimpleAssignment(
+				&ast.SelectorExpr{
+					X:   builder.resultIdent,
+					Sel: ast.NewIdent(string(toProp.PropertyName())),
+				},
+				token.ASSIGN,
+				ast.NewIdent(optionId))
+
+			return []ast.Stmt{result}
+		},
+	}
+}
+
+func (builder *convertToArmBuilder) propertyOfSameTypeHandler() propertyConversionHandler {
+
+	return propertyConversionHandler{
+		finder: func(toProp *astmodel.PropertyDefinition, fromType *astmodel.ObjectType) (bool, *astmodel.PropertyDefinition) {
+			fromProp, ok := fromType.Property(toProp.PropertyName())
+			if ok && toProp.PropertyType().Equals(fromProp.PropertyType()) {
+				return true, fromProp
+			}
+			return false, nil
+		},
+		matchHandler: func(toProp *astmodel.PropertyDefinition, fromProp *astmodel.PropertyDefinition) []ast.Stmt {
+			result := astbuilder.SimpleAssignment(
+				&ast.SelectorExpr{
+					X:   builder.resultIdent,
+					Sel: ast.NewIdent(string(toProp.PropertyName())),
+				},
+				token.ASSIGN,
+				&ast.SelectorExpr{
+					X:   builder.receiverIdent,
+					Sel: ast.NewIdent(string(fromProp.PropertyName())),
+				})
+
+			return []ast.Stmt{result}
+		},
+	}
+}
+
+func (builder *convertToArmBuilder) propertyOfDifferentTypeHandler() propertyConversionHandler {
+
+	return propertyConversionHandler{
+		finder: func(toProp *astmodel.PropertyDefinition, fromType *astmodel.ObjectType) (bool, *astmodel.PropertyDefinition) {
+			fromProp, ok := fromType.Property(toProp.PropertyName())
+			if ok && !toProp.PropertyType().Equals(fromProp.PropertyType()) {
+				return true, fromProp
+			}
+			return false, nil
+		},
+		matchHandler: func(toProp *astmodel.PropertyDefinition, fromProp *astmodel.PropertyDefinition) []ast.Stmt {
+			destination := &ast.SelectorExpr{
+				X:   builder.resultIdent,
+				Sel: ast.NewIdent(string(toProp.PropertyName())),
+			}
+			source := &ast.SelectorExpr{
+				X:   builder.receiverIdent,
+				Sel: ast.NewIdent(string(fromProp.PropertyName())),
+			}
+
+			return builder.toArmComplexPropertyConversion(
+				source,
+				destination,
+				toProp.PropertyType(),
+				string(toProp.PropertyName()),
+				nil,
+				assignmentHandlerAssign)
+		},
+	}
+}
+
+//////////////////////////////////////////////////////////////////////////////////
+// Complex property conversion (for when properties aren't simple primitive types)
+//////////////////////////////////////////////////////////////////////////////////
+
+func (builder *convertToArmBuilder) toArmComplexPropertyConversion(
```

Comment:
This function and the one in the other conversion helper file are basically identical.
The functions being called, with the exception of the TypeName handler, are _almost_ (but not quite) identical. 

I'm not sure if it's worth trying to unify them as the reason for the slight asymmetry is that the code in golang is naturally a bit asymmetric. With some more thinking though we might be able to come up with a way to make it symmetric.